server/personality/personality.service.ts

```diff
@@ -79,33 +79,51 @@ export class PersonalityService {
      * @param personality PersonalityBody received of the client.
      * @returns Return a new personality.
      */
-    create(personality) {
+    async create(personality) {
         try {
-            personality.slug = slugify(personality.name, {
-                lower: true,     // convert to lower case, defaults to `false`
-                strict: true     // strip special characters except replacement, defaults to `false`
-            })
-            const newPersonality = new this.PersonalityModel(personality);
-            this.logger.log(
-                `Attempting to create new personality with data ${personality}`
-            );
-
-            const user = this.req.user
-
-            const history = this.history.getHistoryParams(
-                newPersonality._id,
-                TargetModel.Personality,
-                user,
-                HistoryType.Create,
-                personality
-            )
-
-            this.history.createHistory(history)
-
-            return newPersonality.save();
+            const personalityExisted = await this.getDeletedPersonalityByWikidata(personality.wikidata)
+            if(personalityExisted) {
+                return personalityExisted
```

Comment:
Do the restore in this line of code, to decouple the procedure from the `get` operation. I mean that the `getDeletedPersonalityByWikidata` should just return the personality if it exists and is deleted and not restored.